app/controllers/session.rb

```diff
@@ -0,0 +1,20 @@
+get '/session/new' do
+  erb :'/session/login'
+end
+
+
+post '/session' do
+  user = User.authenticate(params[:user])
+  if user
+    session['user_id'] = user.id
```

Comment:
Heads up: all of your other references to session are to `session[:user_id]`, not to `session['user_id']` as on this line. This _may_ matter; if you are experiencing issues with logging in, this may be why.